app/javascript/mastodon/components/autosuggest_textarea.js

```diff
@@ -17,11 +17,11 @@ const textAtCursorMatchesToken = (str, caretPosition) => {
     word = str.slice(left, right + caretPosition);
   }
 
-  if (!word || word.trim().length < 2 || word[0] !== '@') {
+  if (!word || word.trim().length < 2 || word[0] !== prefix) {
     return [null, null];
   }
 
-  word = word.trim().toLowerCase().slice(1);
```

Comment:
I thought the hashtag would distinguish between uppercase and lowercase letters.
I found it to be all lowercase so fixed it.